src/FEValues/interface_values.jl

```diff
@@ -0,0 +1,582 @@
+
+Return the number of quadrature points on the interface for the current (most recently
+[`reinit!`](@ref)ed) interface.
+"""
+getnquadpoints(iv::InterfaceValues) = getnquadpoints(iv.here)
+
+@propagate_inbounds function getdetJdV(iv::InterfaceValues, q_point::Int)
+    return getdetJdV(iv.here, q_point)
+end
+
+"""
+    reinit!(
+        iv::InterfaceValues,
+        cell_here::AbstractCell, coords_here::AbstractVector{Vec{dim, T}}, face_here::Int,
+        cell_there::AbstractCell, coords_there::AbstractVector{Vec{dim, T}}, face_there::Int
+    )
+
+Update the [`InterfaceValues`](@ref) for the interface between `cell_here` (with cell
+coordinates `coords_here`) and `cell_there` (with cell coordinates `coords_there`).
+`face_here` and `face_there` are the (local) face numbers for the respective cell.
+"""
+function reinit!(
+        iv::InterfaceValues,
+        cell_here::AbstractCell, coords_here::AbstractVector{Vec{dim, T}}, face_here::Int,
+        cell_there::AbstractCell, coords_there::AbstractVector{Vec{dim, T}}, face_there::Int
+    ) where {dim, T}
+
+    # reinit! the here side as normal
+    reinit!(iv.here, coords_here, face_here)
+    dim == 1 && return reinit!(iv.there, coords_there, face_there)
+    # Transform the quadrature points from the here side to the there side
+    iv.there.current_face[] = face_there
+    interface_transformation = InterfaceTransformation(cell_here, cell_there, face_here, face_there)
+    quad_points_a = getpoints(iv.here.qr, face_here)
+    quad_points_b = getpoints(iv.there.qr, face_there)
+    transform_interface_points!(quad_points_b, quad_points_a, interface_transformation)
+    @boundscheck checkface(iv.there, face_there)
+    # TODO: This is the bottleneck, cache it?
+    @assert length(quad_points_a) <= length(quad_points_b)
+    # Re-evalutate shape functions in the transformed quadrature points
+    for qp in 1:length(quad_points_a), i in 1:getnbasefunctions(iv.there)
+        iv.there.dNdξ[i, qp, face_there], iv.there.N[i, qp, face_there] = shape_gradient_and_value(iv.there.func_interp, quad_points_b[qp], i)
+    end
+    for qp in 1:length(quad_points_a), i in 1:getngeobasefunctions(iv.there)
+        iv.there.dMdξ[i, qp, face_there], iv.there.M[i, qp, face_there] = shape_gradient_and_value(iv.there.geo_interp, quad_points_b[qp], i)
+    end
+    # reinit! the "there" side
+    reinit!(iv.there, coords_there, face_there)
+    return iv
+end
+
+"""
+    getnormal(iv::InterfaceValues, qp::Int; here::Bool=true)
+
+Return the normal vector in the quadrature point `qp` on the interface. If `here = true`
+(default) the outward normal to the "here" element is returned, otherwise the outward normal
+to the "there" element.
+"""
+function getnormal(iv::InterfaceValues, qp::Int; here::Bool=true)
+    # TODO: Does it make sense to allow the kwarg here? You can juse negate the vector
+    #       yourself since getnormal(iv, qp; here=false) == -getnormal(iv, qp; here=true).
```

Comment:
Yes. I think on moving domains things can get a bit nastier. But I also do not know how to deal with these.